### DISCUSSION TITLE
Issues with grouping feature

### DISCUSSION THREAD

**Discussion by sthenos** [jamulussoftware/jamulus#1908] [Ideas]
So I got an opportunity to use Jamulus load and save configurations during last nights EHB show.   It worked well for loading and saving a big mix (45 singers in a choir).   The reason I had to use it was because the big choir started the show so I had a good mix for them, then all 45 members individually performed in a number of quartets so I had to alter their levels significantly, then they came together at the end so I wanted to restore my previous big mix.  That worked well, but I also heavily utilised the grouping feature.   This normally works well but there are two flaws:

1.  You can't easily ungroup all 45 people.   I needed to be able to do this.  I was hoping that using the "Clear all stored solo and mute settings" would do it, but it doesn't  I guess the clue is in the name, but I think it really needs to reset pan plus grouping settings too.
2.   When you group it works well on faders, but it does when when you click the pan buttons.  I believe it should just as it would on a group fader on a real desk.

**Comment by wschell** [comment]
I see at least one inconsistency when working with groups: Shift-Fader moves a single fader, but Shift-Pan resets it to neutral. So if grouping works on pan (what I would prefer too), it should behave the same as faders. And grouping should also work for toggling Solo and Mute.
Without any modifier key (Shift, Ctrl, Alt), change settings for whole group.
With modifier 1, change settings only for this member.
With modifier 2, reset fader (to new client level) or pan and toggle mute or solo only for this member.

**Comment by gilgongo** [comment]
Hi @sthenos thanks for this - we'll move it to a discussion until there's an agreement on how to approach the issue.

**Reply by henkdegroot** [reply]
Seems to come up as a regular idea/request:
https://github.com/jamulussoftware/jamulus/discussions/1373

**Reply by sthenos** [reply]
yup this was very frustrating when running a large show.  I'm sure the casual jamulus user or band probably would never use the feature, but as more and more choirs and large ensembles use jamulus it's important to make it easy.  It took me about 2 minutes to un-group everyone once I had grouped them and setting the pan on 50 people takes forever, if it was bundled with the moving of the fader and potentially mute and solo buttons then it would be consistent.

**Comment by jp8** [comment]
Is there a reason for rows and groups to be separate concepts ?  If they were merged, then each group would always get its own row.

I guess what I am asking is...  does anyone use both the groups and the rows functionality, for separate things ?

I am imagining settings like

"group by instrument" - one setting puts the entire choir in a useful arrangement
"group by name prefix" - when a musician changes their name prefix, they jump to the appropriate row
"group by city" - then the city field could be used instead of a name prefix (i know, but anyway it adds some flexibility)

This would allow some visual controls (mute, fader, pan) per group, at the head of each row.
